src/main/kotlin/org/jetbrains/research/libsl/nodes/helpers/ExpressionDumper.kt

```diff
@@ -104,10 +104,24 @@ object ExpressionDumper {
 
     private fun dumpCharacterLiteral(expression: CharacterLiteral): String {
         val value = expression.value
-        val str = String(value.toString().toByteArray(Charsets.UTF_8))
+        val str = escapeCharStringRepresentation(value)
         return "\'${str}\'"
     }
 
+    private fun escapeCharStringRepresentation(value: Char): String {
+        return when (value) {
+            '\n' -> "\\n"
+            '\r' -> "\\r"
+            '\t' -> "\\t"
+            '\b' -> "\\b"
+            '\'' -> "\\'"
+            '\"' -> "\\\""
+            '\\' -> "\\\\"
+            '\u0000' -> "\\0"
+            else -> String(value.toString().toByteArray(Charsets.UTF_8))
+        }
```

Comment:
I suggest implementing a bit more standardized approach here:
* Use "pretty" variants for a specific set of characters.
* Convert to Unicode code points (like `\u1234`) characters that are either control characters or outside of the "Basic Latin" group of symbols.
* Print "Basic Latin" characters as is.

Reply:
Fixed;